compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenLibraryManager.cpp

```diff
@@ -55,6 +55,14 @@ static FailureOr<ModuleOp> getOrParseTransformLibraryModuleImpl(
   return parsedLibrary.get();
 }
 
+std::optional<ModuleOp> IREECodegenDialect::getLoneTransformLibraryModule() {
+  std::lock_guard<std::mutex> guard(libraryMutex);
+  if (libraryModules.size() != 1) {
+    return std::nullopt;
+  }
+  return libraryModules[*libraryModules.keys().begin()].get();
```

Comment:
I think you can do `libraryModules.being()->first;` to avoid hashing

Reply:
If `libraryModules` is of size 1, why even make it a `StringMap`?

Reply:
That was what I wondered above as well. The reason I initially did it was so we could load multiple different modules (e.g. different module per target), but I realized for this change there isn't a way to know which library module to look for, so I cheated by looking for a single module.

After further thought, the StringMap + ability to retain multiple modules is likely important. Users won't want to precombine all tuning specs for all targets and instead maybe pass it in as a target specific module or something. In those cases we need the map, but we can restrict it to a single module per backend. What I'm going to do is add a discardable field to the parent module similar to `iree_codegen.tuning_spec_mlirbc` that stores the filename to lookup.

Reply:
This helper is dropped now.